core/src/main/java/de/javakaffee/web/msm/StorageClientFactory.java

```diff
@@ -78,9 +87,30 @@ static MemcachedClient createCouchbaseClient(final MemcachedNodesManager memcach
             }
         }
 
+        protected static MemcachedNodeURLBasedConnectionFactory createMemcachedNodeURLBasedConnectionFactory() {
+            try {
+                Class.forName("net.spy.memcached.ClientMode");
+                return  Class.forName("de.javakaffee.web.msm.MemcachedElasticConnectionFactory").asSubclass(MemcachedNodeURLBasedConnectionFactory.class).newInstance();
+            } catch (final ClassNotFoundException e) {
+                try {
+                    return Class.forName("de.javakaffee.web.msm.MemcachedConnectionFactory").asSubclass(MemcachedNodeURLBasedConnectionFactory.class).newInstance();
+                } catch (final Exception ex)
+                {
+                    throw new RuntimeException(ex);
+                }
+            } catch (final Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
         static ConnectionFactory createConnectionFactory(final MemcachedNodesManager memcachedNodesManager,
                                                             final ConnectionType connectionType, final String memcachedProtocol, final String username, final String password, final long operationTimeout,
                                                             final long maxReconnectDelay, final Statistics statistics ) {
+            boolean isClientDynamicMode = false;
+            if (memcachedNodesManager.getMemcachedNodes().contains(".cfg."))
+            {
+                isClientDynamicMode=true;
```

Comment:
Why is the clientMode set to "Dynamic" statically? And why is this coupled to `if (memcachedNodesManager.getMemcachedNodes().contains(".cfg."))`?

I'd expect that the environment variable / system property `client.mode` is evaluated...

And as already said, determining the clientMode should be done inside the `MemcachedElastiCacheConnectionFactory`.